Integrate Foundation for Apps framework

myabc opened the pull request:
https://community.openproject.org/work_packages/17751

Begins a major CSS refactor.
- [x] Layout fixes (major issues)
- [x] News fixes
- [x] Buttons
- [x] Update OP themes
- [x] Update OP plugins
  - [x] Backlogs
  - [x] Costs
  - [x] Reporting (Engine)
  - [x] My Project Page
  - [x] Mobile OTP
  - [x] Meetings
- [x] Typography (normalisation)
- [x] Basic Form fixes
  - [x] General
- [x] Pagination fixes

Moved to other PRs:
- Filter design improvements
- Button groups, segmented buttons (in toolbar)
- Progress bars
- Form rework:
  - File upload
  - Modals

myabc committed:
Remove own font-face mixin in favour of Bourbon's

This partially reverts e5f5b7ac230cbe767cc113021ec9740066220b2f

Signed-off-by: Alex Coles <alex@alexbcoles.com>

myabc committed:
Add Foundation for Apps

Signed-off-by: Alex Coles <alex@alexbcoles.com>

myabc committed:
Add foundation-apps to Sass load path (Sprockets)

Signed-off-by: Alex Coles <alex@alexbcoles.com>

myabc committed:
Add foundation-apps to Sass load path (Gulp)

Signed-off-by: Alex Coles <alex@alexbcoles.com>

myabc committed:
Remove extraneous @import directives in Sass files

Signed-off-by: Alex Coles <alex@alexbcoles.com>

myabc committed:
Move global Sass partials to open_project_global

Avoid potential naming collision with Foundation for Apps.

Signed-off-by: Alex Coles <alex@alexbcoles.com>

myabc committed:
Bump Sass to latest 3.4.x

Signed-off-by: Alex Coles <alex@alexbcoles.com>

myabc committed:
Integrate Foundation global styles

Signed-off-by: Alex Coles <alex@alexbcoles.com>

TeatroIO commented:
I've prepared a stage. <a href='http://feature-foundation-apps-framework.opf-openproject-2af22bf12c7dfdad4ddf.ttrcloud.com' target='_blank'>Click to open.</a>

myabc committed:
Remove border-style from #main

Signed-off-by: Alex Coles <alex@alexbcoles.com>

myabc committed:
Remove reset in favour of Foundation's normalise

Signed-off-by: Alex Coles <alex@alexbcoles.com>

myabc committed:
Dasherize all Sass variables

Signed-off-by: Alex Coles <alex@alexbcoles.com>

myabc committed:
Dasherize all Sass mixins

Signed-off-by: Alex Coles <alex@alexbcoles.com>

myabc committed:
Living Style Guide: Fix variable name

[ci skip]

Signed-off-by: Alex Coles <alex@alexbcoles.com>

myabc committed:
Revert "Living Style Guide: Document OpenProject colors"

This reverts commit 91b6bc130dc462595b22c909f6ce41a265418797.

myabc committed:
Remove unused OpenProject colour variables

This partially reverts 2504a873f6cd7405314b595c6ead1b4c96568baa

Signed-off-by: Alex Coles <alex@alexbcoles.com>

myabc committed:
Begin to rationalise Sass colour variables

Signed-off-by: Alex Coles <alex@alexbcoles.com>

myabc committed:
Remove border-radius mixin; use property directly

Signed-off-by: Alex Coles <alex@alexbcoles.com>

myabc committed:
Rework buttons to use Foundation buttons mixins

Signed-off-by: Alex Coles <alex@alexbcoles.com>

myabc committed:
Switch to released foundation-apps

Signed-off-by: Alex Coles <alex@alexbcoles.com>

myabc committed:
Add newly generated Foundation _settings.scss

Signed-off-by: Alex Coles <alex@alexbcoles.com>

myabc committed:
Disable printing of all Foundation CSS classes

Signed-off-by: Alex Coles <alex@alexbcoles.com>

myabc committed:
Remove overly-general override for buttons

Signed-off-by: Alex Coles <alex@alexbcoles.com>

myabc committed:
Ensure #main always uses 100% width

and not just in in absolute layout mode.

Signed-off-by: Alex Coles <alex@alexbcoles.com>

myabc committed:
Adjust search field for border-box, Foundation

myabc committed:
Temp. remove 'magnifier' from search field

This rectifies issues with the main layout.

myabc committed:
Enable printing of Foundation typography classes

Signed-off-by: Alex Coles <alex@alexbcoles.com>

myabc committed:
Remove all non-variablised pixel font sizes

While drastic this is a necessary step towards refactoring: moving
from arbitrary font-sizes to typography based on ratios.

Signed-off-by: Alex Coles <alex@alexbcoles.com>

myabc committed:
Remove legacy typography styles

Signed-off-by: Alex Coles <alex@alexbcoles.com>

myabc committed:
Fix breadcrumb margins, paddings

Signed-off-by: Alex Coles <alex@alexbcoles.com>

myabc committed:
Fix select rendering: remove height from own mixin

Signed-off-by: Alex Coles <alex@alexbcoles.com>

myabc committed:
Rename classes: single button class with modifiers

Signed-off-by: Alex Coles <alex@alexbcoles.com>

myabc committed:
Refactor (DRY up) Sass for button styles

Signed-off-by: Alex Coles <alex@alexbcoles.com>

myabc committed:
Add button size modifiers

Signed-off-by: Alex Coles <alex@alexbcoles.com>

myabc committed:
Living Style Guide: Document button size modifiers

Reorganise button documentation.

[ci skip]

Signed-off-by: Alex Coles <alex@alexbcoles.com>

myabc committed:
Remove (apparently) unused legacy button styles

Signed-off-by: Alex Coles <alex@alexbcoles.com>

myabc committed:
Apply button styles to most buttons in Rails views

Signed-off-by: Alex Coles <alex@alexbcoles.com>

myabc committed:
Eliminate menu custom button styles

Instead use standard button with round modifier.

Signed-off-by: Alex Coles <alex@alexbcoles.com>

myabc committed:
Add, document round button modifier

Signed-off-by: Alex Coles <alex@alexbcoles.com>

myabc committed:
Adjust top menu login fields for border-box

Signed-off-by: Alex Coles <alex@alexbcoles.com>

myabc committed:
Reduce button vertical padding slightly globally

myabc committed:
Merge branch 'dev' into feature/foundation-apps-framework

Signed-off-by: Alex Coles <alex@alexbcoles.com>

Conflicts:
	frontend/bower.json

myabc committed:
'Namespace' date/time directives with op- prefix

Avoid collision with native HTML <time> element.

Signed-off-by: Alex Coles <alex@alexbcoles.com>

myabc committed:
Rework time_tag Helper to generate HTML <time> tag

Signed-off-by: Alex Coles <alex@alexbcoles.com>

myabc committed:
Move news-related classes out of legacy styles

Signed-off-by: Alex Coles <alex@alexbcoles.com>

myabc committed:
Living Style Guide: Document news-related styles

[ci skip]

Signed-off-by: Alex Coles <alex@alexbcoles.com>

myabc committed:
Amend MenuItem to append -menu-item class suffix

Keep CSS class names more tightly scoped.

Signed-off-by: Alex Coles <alex@alexbcoles.com>

myabc commented:
@ulferts @0xF013 this is ready for review. I've had to shrink the scope and break some subtasks out into other PRs/WPs.

ulferts commented on app/assets/stylesheets/default.css.sass:
```diff
@@ -33,12 +33,17 @@
 //= require_self
 //= require legacy/main
```
How does this relate to https://github.com/opf/openproject/pull/2353? Will the menu on wiki et. al. when using themes be lost again.

myabc replied:
@ulferts the changes from #2353 didn't make it into dev.

ulferts replied:
I understand that. That will then be addressed in a separate PR I guess?

myabc replied:
@ulferts yes. Better theming is beyond the scope of this PR.

ulferts commented:
The forms in various places are broken:

WP Filter:
![image](https://cloud.githubusercontent.com/assets/617519/5507241/c642d472-87a1-11e4-8041-7fab0218f3ba.png)

Projects administration:
![image](https://cloud.githubusercontent.com/assets/617519/5507251/d501b2b2-87a1-11e4-9b87-a521db6d4d6f.png)

Users administration:
![image](https://cloud.githubusercontent.com/assets/617519/5507271/0eb243aa-87a2-11e4-85e2-71c1c2d750a6.png)

Timelines
![image](https://cloud.githubusercontent.com/assets/617519/5507297/38a25f10-87a2-11e4-8369-8b26c8e7c19a.png)

Timelines administration:
![image](https://cloud.githubusercontent.com/assets/617519/5507309/4e4e620a-87a2-11e4-9aca-2e6051de65af.png)

![image](https://cloud.githubusercontent.com/assets/617519/5507318/61dfe82a-87a2-11e4-861d-38d0f10fa687.png)

Calendar:
![image](https://cloud.githubusercontent.com/assets/617519/5507329/6d788426-87a2-11e4-9707-8c283249a44f.png)

myabc commented:
> in various places are broken:

see my comment on the WP. I don't want to address form sizing (which would mean adding a bunch of ad-hoc hacks/overrides) until I address text sizing overall.

ulferts commented:
Action menus have discs before each item:

![image](https://cloud.githubusercontent.com/assets/617519/5507406/1e0795ca-87a3-11e4-9641-65b5ef04ef32.png)

myabc committed:
Remove list styling from legacy action menu

Signed-off-by: Alex Coles <alex@alexbcoles.com>

myabc commented:
@ulferts 

> Action menus have discs before each item:

should be fixed in 207a396.

myabc committed:
Merge branch 'feature/foundation-apps-further-refactoring' into feature/foundation-apps-framework

myabc committed:
Remove unnecessary units from CSS 0 values

Signed-off-by: Alex Coles <alex@alexbcoles.com>

myabc committed:
Enable printing of Foundation grid classes

Signed-off-by: Alex Coles <alex@alexbcoles.com>

myabc committed:
Remove unnecessary unit from another CSS 0 value

Signed-off-by: Alex Coles <alex@alexbcoles.com>

myabc committed:
Rework work package filters using CSS flexboxes

Switch markup from table to (slightly more semantic) `<ul>`.

Signed-off-by: Alex Coles <alex@alexbcoles.com>

myabc committed:
Merge branch 'dev' into feature/foundation-apps-framework

myabc committed:
Remove remaining vendor prefixes from Sass/CSS

Signed-off-by: Alex Coles <alex@alexbcoles.com>

myabc committed:
In-line heading Sass mixins

Signed-off-by: Alex Coles <alex@alexbcoles.com>

myabc committed:
Remove stray variable in mixins Sass partial

Signed-off-by: Alex Coles <alex@alexbcoles.com>

myabc committed:
Remove unnecessary html-body-spacing Sass mixin

Foundation's settings (with default `$include-html-classes: true`)
will set these sane defaults for us.

Signed-off-by: Alex Coles <alex@alexbcoles.com>

myabc committed:
Remove unnecessary reset on h1

Foundation's typography component takes care of this.

myabc committed:
Consolidate wiki heading styles

Signed-off-by: Alex Coles <alex@alexbcoles.com>

myabc committed:
Add class for selects, select size modifiers

Signed-off-by: Alex Coles <alex@alexbcoles.com>

myabc committed:
Living Style Guide: Document select size modifiers

[ci skip]

Signed-off-by: Alex Coles <alex@alexbcoles.com>

myabc committed:
Living Style Guide: Document text field sizes

[ci skip]

Signed-off-by: Alex Coles <alex@alexbcoles.com>

myabc committed:
Fix filters horiz. layout on admin projects page

Signed-off-by: Alex Coles <alex@alexbcoles.com>

myabc committed:
Merge branch 'dev' into feature/foundation-apps-framework

Signed-off-by: Alex Coles <alex@alexbcoles.com>

Conflicts:
	app/assets/stylesheets/_reset.sass
	app/assets/stylesheets/global/_mixins.sass
	app/assets/stylesheets/global/_variables.sass

myabc committed:
Fix label spacing in Administration Settings

Signed-off-by: Alex Coles <alex@alexbcoles.com>

myabc committed:
Remove unnecessary font-family, -size overrides

Where possible, let font properties be inherited.

Signed-off-by: Alex Coles <alex@alexbcoles.com>

myabc committed:
In-line breadcrumb font mixins

Signed-off-by: Alex Coles <alex@alexbcoles.com>

myabc committed:
In-line main menu font mixins

Signed-off-by: Alex Coles <alex@alexbcoles.com>

myabc committed:
In-line header-fonts mixin

Signed-off-by: Alex Coles <alex@alexbcoles.com>

myabc committed:
Combine default font mixins into one mixin

Signed-off-by: Alex Coles <alex@alexbcoles.com>

myabc committed:
Import Foundation _settings.scss after own globals

Signed-off-by: Alex Coles <alex@alexbcoles.com>

myabc committed:
Remove font-family, -size overrides on headings

Signed-off-by: Alex Coles <alex@alexbcoles.com>

myabc committed:
Replace own variables with Foundation's: global

Signed-off-by: Alex Coles <alex@alexbcoles.com>

myabc committed:
Remove unnecessary colour overrides on body

Signed-off-by: Alex Coles <alex@alexbcoles.com>

myabc committed:
Replace own variables with Foundation's: typography

Signed-off-by: Alex Coles <alex@alexbcoles.com>

myabc committed:
Convert timelines styles to Sass

Signed-off-by: Alex Coles <alex@alexbcoles.com>

myabc committed:
Adjust size of selects used in timelines toolbar

Signed-off-by: Alex Coles <alex@alexbcoles.com>

myabc committed:
Adjust legacy action menu for new base font size

Signed-off-by: Alex Coles <alex@alexbcoles.com>

myabc committed:
Add class for text fields with size modifiers

Signed-off-by: Alex Coles <alex@alexbcoles.com>

myabc committed:
Fix filters horiz. layout on users page

Signed-off-by: Alex Coles <alex@alexbcoles.com>

myabc committed:
Fix year/month filter horiz. layout on calendar

Signed-off-by: Alex Coles <alex@alexbcoles.com>

myabc committed:
Move .box, .tabular classes out of legacy styles

Signed-off-by: Alex Coles <alex@alexbcoles.com>

myabc committed:
Move .legacy-pagination class out of legacy styles

Signed-off-by: Alex Coles <alex@alexbcoles.com>

myabc committed:
Living Style Guide: Document legacy pagination

[ci skip]

Signed-off-by: Alex Coles <alex@alexbcoles.com>

myabc commented:
@ulferts will address Rubocop warnings in TU tests in a separate PR.

myabc committed:
Move table related classes out of legacy styles

Signed-off-by: Alex Coles <alex@alexbcoles.com>

myabc committed:
Remove width overrides for user form fields

Signed-off-by: Alex Coles <alex@alexbcoles.com>

myabc committed:
Tweak padding on tabular forms

Signed-off-by: Alex Coles <alex@alexbcoles.com>

myabc committed:
Fix field instruction spacing in Projects form

Signed-off-by: Alex Coles <alex@alexbcoles.com>

myabc committed:
Fix field instruction spacing in Users form

Signed-off-by: Alex Coles <alex@alexbcoles.com>

myabc committed:
Remove width override for textarea.wiki-edit

Signed-off-by: Alex Coles <alex@alexbcoles.com>

myabc committed:
Remove stray variables in toolbar partial

Signed-off-by: Alex Coles <alex@alexbcoles.com>

myabc committed:
Rework toggledMultiselect to use postfix labels

Signed-off-by: Alex Coles <alex@alexbcoles.com>

myabc committed:
Right-align work package filters add filter item

Signed-off-by: Alex Coles <alex@alexbcoles.com>

myabc committed:
Fix calendar filter layout, use postfix labels

Signed-off-by: Alex Coles <alex@alexbcoles.com>

myabc committed:
Fix typo in ng-switch in queryFilters template

Signed-off-by: Alex Coles <alex@alexbcoles.com>

myabc committed:
Remove duplicate labels in queryFilters template

Signed-off-by: Alex Coles <alex@alexbcoles.com>

myabc committed:
Fix feature specs for markup changes in c9acbfbd

Signed-off-by: Alex Coles <alex@alexbcoles.com>

myabc committed:
Bump foundation-apps to 1.0.2

Signed-off-by: Alex Coles <alex@alexbcoles.com>

myabc committed:
Fix layout of datepicker year/month selects

Signed-off-by: Alex Coles <alex@alexbcoles.com>

myabc commented:
@ulferts I should have now addressed all of the issues you highlighted, with the exception of _Timelines administration page_.
- Work package filters have received a big overhaul.
- Other pages/views have more minimal changes:
  - Projects administration
  - Users administration
  - Timelines
  - Calendar

Hopefully, there'll be time to improve these views in a future PR.

myabc committed:
Remove conflicting input, textarea styles

Signed-off-by: Alex Coles <alex@alexbcoles.com>

ulferts commented:
I like the new filter layout for work packages!

I have looked through the application and made screenshots of everything that looked weird to me. Lets talk about what needs to be fixed before merging:

Positioning of the query menu:

![image](https://cloud.githubusercontent.com/assets/617519/5660182/6650393e-971d-11e4-8318-ff489543c059.png)

WP Query configuration modal:

![image](https://cloud.githubusercontent.com/assets/617519/5660209/b00e9cb4-971d-11e4-998c-da2db064d16a.png)

WP sort modal:

![image](https://cloud.githubusercontent.com/assets/617519/5660216/be344140-971d-11e4-9c39-01c4470bbaaf.png)

Dates on activity tab of details pane:

![image](https://cloud.githubusercontent.com/assets/617519/5659679/c0a96e16-9716-11e4-8a27-4d771c5bd81a.png)

Adding relations on old wp show page:

![image](https://cloud.githubusercontent.com/assets/617519/5659702/4e7e8e2e-9717-11e4-9696-8978faee243f.png)

Adding watchers on old wp show page:

![image](https://cloud.githubusercontent.com/assets/617519/5659709/67e16fb2-9717-11e4-9c5f-53ae8a5658b7.png)

Timeline edit page:

![image](https://cloud.githubusercontent.com/assets/617519/5659745/0b7e49b0-9718-11e4-9a9b-7a5aa3c8d820.png)

Calendar filtering:

![image](https://cloud.githubusercontent.com/assets/617519/5661602/054e9426-972e-11e4-8ac3-773b1deed542.png)

Repository revision jump box:

![image](https://cloud.githubusercontent.com/assets/617519/5659772/61eee3cc-9718-11e4-87f8-e4ba338341c4.png)

Repository revision diff view selector:

![image](https://cloud.githubusercontent.com/assets/617519/5659793/7f7391a4-9718-11e4-90e7-59a9cb1fcdc6.png)

Wiki create new child page form:

![image](https://cloud.githubusercontent.com/assets/617519/5659801/9849c248-9718-11e4-82b6-5f6b184daa5a.png)

New status reporting form:

![image](https://cloud.githubusercontent.com/assets/617519/5659808/c0716a14-9718-11e4-873f-b1c0a14d89ce.png)

New dependency form:

![image](https://cloud.githubusercontent.com/assets/617519/5659814/d88b4868-9718-11e4-9d83-5d038ef82418.png)

Project settings module selection:

![image](https://cloud.githubusercontent.com/assets/617519/5659830/1310b3a6-9719-11e4-8e89-e4548b531c7b.png)

Admin project filter:

![image](https://cloud.githubusercontent.com/assets/617519/5661632/5ad6a12c-972e-11e4-95be-7a2a9cf9a382.png)

Admin user filtering:

Same as above (buttons not aligned)

Admin workflows role/type selection:

![image](https://cloud.githubusercontent.com/assets/617519/5659903/23515b52-971a-11e4-9a9c-4024e8db10a6.png)

Admin custom field edit:

![image](https://cloud.githubusercontent.com/assets/617519/5659923/39627872-971a-11e4-906e-24bf8d4220fe.png)

Admin colors form:

![image](https://cloud.githubusercontent.com/assets/617519/5659949/7afd2ce6-971a-11e4-88cd-a24d5b8300ef.png)

My page edit block selection (has become unusable because of the overlapping):

![image](https://cloud.githubusercontent.com/assets/617519/5659960/9d7b6a12-971a-11e4-9c50-8bc8f8481f74.png)

Login screen:

![image](https://cloud.githubusercontent.com/assets/617519/5660075/20654988-971c-11e4-8d83-9ace48091c75.png)

Cost reporting plugin filter:

![image](https://cloud.githubusercontent.com/assets/617519/5660090/47497b46-971c-11e4-941d-b454d92b2e28.png)

As a general note: we should probably limit the max width of input fields as they otherwise tend to look strange.

myabc commented:
@ulferts 

RE:

> Timeline toolbar:
> Calendar filtering:
> Admin project filter:
> Admin user filtering:

Do you have the latest commits?

RE:

> Filters on activity page:
> Admin group user selection:
> Admin custom field edit:

Please make sure you're on Foundation for Apps 1.0.2. This new release [improves checkbox styling](https://github.com/zurb/foundation-apps/blob/master/CHANGELOG.md#version-102).

myabc commented:
@ulferts 

> WP sort modal:

Only seeing this at certain breakpoints. Would prefer to migrate to ui-select before fixing this.

myabc committed:
Fix member roles checkbox spacing in proj.settings

Signed-off-by: Alex Coles <alex@alexbcoles.com>

myabc committed:
Fix checkbox spacing in group tab of users form

Signed-off-by: Alex Coles <alex@alexbcoles.com>

ulferts commented:
@myabc sorry again for not having used the latest commit. I'll edit my initial comment and remove everything that no longer applies.

myabc committed:
[wip] Adjust withDropdown vertical positioning

Signed-off-by: Alex Coles <alex@alexbcoles.com>

myabc committed:
Remove various box-sizing overrides

Foundation will apply the border-box box-sizing on all elements (with
the exception of `<hr>` and `<input type="search">`).

Signed-off-by: Alex Coles <alex@alexbcoles.com>

myabc committed:
Fix query dropdown padding, magnifier position

Signed-off-by: Alex Coles <alex@alexbcoles.com>

myabc committed:
Undo withDropdown changes breaking query dropdown

This partially reverts a56615f9fa4e99dd080cde8ba4d8c8145f96f7bd

Signed-off-by: Alex Coles <alex@alexbcoles.com>

myabc committed:
Revert "Temp. remove 'magnifier' from search field"

This reverts commit 6590e8db5c945867f5f1502e2e3d26730286a5a2.

myabc committed:
Fix positioning of search field magnifier

Signed-off-by: Alex Coles <alex@alexbcoles.com>

myabc committed:
Fix broken legacy action menu HTML on My Page

Signed-off-by: Alex Coles <alex@alexbcoles.com>

myabc committed:
Fix add page block horiz. layout on My Page

Signed-off-by: Alex Coles <alex@alexbcoles.com>

myabc commented:
@ulferts issues in plugins should now have been addressed.

ulferts commented:
Sorry for spotting this so late but the meetings plugin also has one problem:

![image](https://cloud.githubusercontent.com/assets/617519/5703903/a701ba8c-9a6c-11e4-975a-7410c9f7c447.png)

myabc commented:
@ulferts meetings is improved – time/date is still on lines though. This is not something I can fix without further refactoring of our forms CSS.

myabc committed:
Remove list styling from wp activities list

Signed-off-by: Alex Coles <alex@alexbcoles.com>

myabc committed:
Fix excess padding on work package filter fields

Signed-off-by: Alex Coles <alex@alexbcoles.com>

myabc committed:
(Vert.) center align work package filter flexboxes

baseline implementation appears to be buggy in Chrome.

Signed-off-by: Alex Coles <alex@alexbcoles.com>

myabc committed:
Fix pagination item widths for border-box

Signed-off-by: Alex Coles <alex@alexbcoles.com>

myabc committed:
Fix member roles checkbox spacing in group admin

Signed-off-by: Alex Coles <alex@alexbcoles.com>

myabc committed:
Fix field instruction spacing in My Password form

Signed-off-by: Alex Coles <alex@alexbcoles.com>

myabc committed:
Fix label spacing in delete category form

Signed-off-by: Alex Coles <alex@alexbcoles.com>

myabc committed:
Fix field instruction spacing in account forms

Signed-off-by: Alex Coles <alex@alexbcoles.com>

myabc committed:
Fix incorrect args passed to toggle_link helper

Broken in bd22b353

Signed-off-by: Alex Coles <alex@alexbcoles.com>

myabc committed:
Fix layout of add watcher fields on legacy wp view

Signed-off-by: Alex Coles <alex@alexbcoles.com>

myabc committed:
Fix layout of add relation fields on legacy wp view

Signed-off-by: Alex Coles <alex@alexbcoles.com>

myabc commented:
@ulferts I dealt with _most_ but _not all_ of the issues you mentioned. I'd like to break the rest out into separate WP/PRs.

myabc committed:
Merge branch 'dev' into feature/foundation-apps-framework

myabc committed:
Reduce form element padding slightly globally

Signed-off-by: Alex Coles <alex@alexbcoles.com>

myabc committed:
Merge branch 'dev' into feature/foundation-apps-framework

Signed-off-by: Alex Coles <alex@alexbcoles.com>

Conflicts:
	app/assets/stylesheets/content/_flash_messages.sass

myabc committed:
Improve filters layout on admin projects page

Signed-off-by: Alex Coles <alex@alexbcoles.com>

myabc committed:
Improve filters layout on users page

Signed-off-by: Alex Coles <alex@alexbcoles.com>

myabc committed:
Remove .bold class in favour of <strong>

Signed-off-by: Alex Coles <alex@alexbcoles.com>

myabc committed:
Minor cleanup of unnecessary padding value

Signed-off-by: Alex Coles <alex@alexbcoles.com>

myabc committed:
Remove margin overrides on input, selects

Signed-off-by: Alex Coles <alex@alexbcoles.com>

myabc committed:
Fix property indentation in forms partial

myabc committed:
Fix styled checkboxes by using content-box sizing

Signed-off-by: Alex Coles <alex@alexbcoles.com>

myabc committed:
Normalise indentation in Rails ERB views

Signed-off-by: Alex Coles <alex@alexbcoles.com>

ulferts committed:
fix label+input/select styling

myabc committed:
Replace splitcontent* classes with Foundation grid

Signed-off-by: Alex Coles <alex@alexbcoles.com>

ulferts committed:
remove conditional comment

myabc committed:
Remove a no longer necessary clear element

Signed-off-by: Alex Coles <alex@alexbcoles.com>

myabc committed:
Enable printing of Foundation utilities classes

Signed-off-by: Alex Coles <alex@alexbcoles.com>

myabc committed:
Remove own clearfix classes

Signed-off-by: Alex Coles <alex@alexbcoles.com>

myabc committed:
Add clearfix element after legacy action menu

Foundation's grid classes do not currently clear.

Signed-off-by: Alex Coles <alex@alexbcoles.com>

myabc committed:
Move various component styles out of legacy styles

What remains is more or less a "junk-drawer" of various ad-hoc rules.

Signed-off-by: Alex Coles <alex@alexbcoles.com>

myabc committed:
Recombine misc. legacy styles into single partial

This effectively reverts a64a8672e379a2fca9ec95c9f455e938dcd829c5

Signed-off-by: Alex Coles <alex@alexbcoles.com>

myabc committed:
Merge pull request #2419 from opf/feature/foundation-apps-framework-inline-label

fix label+input/select styling

myabc committed:
Rework work package attributes using CSS flexboxes

Signed-off-by: Alex Coles <alex@alexbcoles.com>

ulferts committed:
rename css class to standard

myabc committed:
Reduce size of wp attribute labels slightly

myabc committed:
Fix wp attribute group headers font size, weight

myabc committed:
Remove unnecessary floats

myabc committed:
Remove old toggler styles (dead code)

Signed-off-by: Alex Coles <alex@alexbcoles.com>

myabc committed:
Simplify panelExpander template markup

Signed-off-by: Alex Coles <alex@alexbcoles.com>

myabc committed:
Remove arrows from panelExpander template

Signed-off-by: Alex Coles <alex@alexbcoles.com>

myabc committed:
Reposition wp attribute group 'show all' toggle

* Rework work package attribute groups with flexboxes.

Signed-off-by: Alex Coles <alex@alexbcoles.com>

myabc committed:
Simplify 'show all' toggle wording

Signed-off-by: Alex Coles <alex@alexbcoles.com>

myabc committed:
Add, document hollow, transparent button modifiers

Signed-off-by: Alex Coles <alex@alexbcoles.com>

myabc committed:
Apply transparent button styling to 'show all'

Signed-off-by: Alex Coles <alex@alexbcoles.com>

myabc committed:
Move wp attribute styles into own partial

Signed-off-by: Alex Coles <alex@alexbcoles.com>

myabc committed:
Tweak 'show all' button margin

Signed-off-by: Alex Coles <alex@alexbcoles.com>

myabc committed:
Living Style Guide: Document single attr. styles

Signed-off-by: Alex Coles <alex@alexbcoles.com>

myabc committed:
Rename attribute-related classes to be generic

Remove `work-package` from prefix: it is envisaged that this styling
can be applied in various views, not just work packages details view.

Signed-off-by: Alex Coles <alex@alexbcoles.com>

myabc committed:
Tweak vertical position of 'show all' button

Signed-off-by: Alex Coles <alex@alexbcoles.com>

myabc committed:
Merge branch 'dev' into feature/foundation-apps-framework

Signed-off-by: Alex Coles <alex@alexbcoles.com>

Conflicts:
	frontend/app/ui_components/index.js
	frontend/app/ui_components/with-dropdown-directive.js
	frontend/public/templates/work_packages.list.html

myabc committed:
Remove hollow button modifier for now

myabc committed:
Adjust transparent button hover border colour

Signed-off-by: Alex Coles <alex@alexbcoles.com>

myabc committed:
Merge branch 'dev' into feature/foundation-apps-framework

myabc committed:
Adjust background for transparent active buttons

Signed-off-by: Alex Coles <alex@alexbcoles.com>

myabc committed:
Remove padding on attribute values

Signed-off-by: Alex Coles <alex@alexbcoles.com>

ulferts committed:
Merge pull request #2421 from opf/feature/17193-work-package-attribute-styling

17193 Work package attribute styling

myabc committed:
Remove unused legacy styles: .square class

Signed-off-by: Alex Coles <alex@alexbcoles.com>

myabc committed:
Remove unused legacy styles: .extras-actions class

Signed-off-by: Alex Coles <alex@alexbcoles.com>

myabc committed:
Remove unused legacy styles: .file-thumbs class

Signed-off-by: Alex Coles <alex@alexbcoles.com>

myabc committed:
Remove unused legacy styles: .question-line class

Signed-off-by: Alex Coles <alex@alexbcoles.com>

myabc committed:
Remove unused styles for #multiple-action-buttons

Signed-off-by: Alex Coles <alex@alexbcoles.com>

myabc committed:
Remove unused legacy styles: .profile-box class

Signed-off-by: Alex Coles <alex@alexbcoles.com>

myabc committed:
Remove unused legacy styles: .action-report class

Signed-off-by: Alex Coles <alex@alexbcoles.com>

myabc committed:
Fix details pane toolbar padding, menu position

Signed-off-by: Alex Coles <alex@alexbcoles.com>

myabc committed:
Remove further stray variables in toolbar partial

Signed-off-by: Alex Coles <alex@alexbcoles.com>

myabc committed:
Quick fix toolbar margin, menu vertical position

/ HT @0xF013

See a56615f9.

Signed-off-by: Alex Coles <alex@alexbcoles.com>

myabc committed:
Living Style Guide: Add missing form classes

[ci skip]

Signed-off-by: Alex Coles <alex@alexbcoles.com>

myabc committed:
Merge branch 'dev' into feature/foundation-apps-framework

Signed-off-by: Alex Coles <alex@alexbcoles.com>

Conflicts:
	frontend/public/templates/work_packages/tabs/overview.html

myabc committed:
Living Style Guide: Add custom header, navigation

[ci skip]

Signed-off-by: Alex Coles <alex@alexbcoles.com>

myabc committed:
Ensure div around 'show all' button never scrolls

myabc committed:
Remove flexbox behaviour from .attributes-group

This is currently unnecessary. Worse, this results in attributes
groups not being rendered (or being rendered with 0 height) on IE11.

Signed-off-by: Alex Coles <alex@alexbcoles.com>

myabc committed:
Remove overflow override on wiki pages

Signed-off-by: Alex Coles <alex@alexbcoles.com>

myabc commented:
@ulferts I've created a separate work package for form refactoring (this will also resolve the issue with upload fields).
